sim/src/main.rs

```diff
@@ -26,20 +26,28 @@ fn main() {
                 .value_name("PLUGIN")
                 .help("Path to the plugin."),
         )
+        .arg(
+            Arg::with_name("random_num_seed")
+                .short("r")
+                .long("random_num_seed")
+                .value_name("RANDOM_NUM_SEED")
+                .help("A seed for all the random routing decisions."),
+        )
         .get_matches();
 
     // Set up library access
     let plugin_str = matches.value_of("plugin");
+    let seed = matches.value_of("random_num_seed");
 
     // Create simulator object.
-    let mut simulator: Simulator = Simulator::new();
+    let mut simulator: Simulator = Simulator::new(seed);
```

Comment:
If this is None, generate seed and print. With that a run can always be reproduced.